src/interfaces/ecpg/preproc/type.c

```diff
@@ -351,7 +357,7 @@ ECPGdump_a_type(FILE *o, const char *name, struct ECPGtype * type, const int bra
 					mmfatal(INDICATOR_NOT_SIMPLE, "indicator for simple data type has to be simple");
 
 				ECPGdump_a_simple(o, name, type->type, str_varchar_one, (arr_str_siz && strcmp(arr_str_siz, "0") != 0) ? arr_str_siz : str_arr_one, struct_sizeof, prefix, 0);
-				if (ind_type != NULL)
+				if (ind_type != NULL && ind_name != NULL)
```

Comment:
Do these places require `&& ind_name != NULL` as well?
https://github.com/arenadata/gpdb/blob/a521853a5c79c1033b296be7689e373f2ccdc281/src/interfaces/ecpg/preproc/type.c#L381
https://github.com/arenadata/gpdb/blob/a521853a5c79c1033b296be7689e373f2ccdc281/src/interfaces/ecpg/preproc/type.c#L401

Reply:
It is not obvious from the code that they are together NULL or not NULL.

Reply:
In `ECPGdump_a_simple()` name is used for `ECPGt_descriptor`:
https://github.com/arenadata/gpdb/blob/a521853a5c79c1033b296be7689e373f2ccdc281/src/interfaces/ecpg/preproc/type.c#L424-L426
And for `default` case as in `ECPGdump_a_type()`:
https://github.com/arenadata/gpdb/blob/a521853a5c79c1033b296be7689e373f2ccdc281/src/interfaces/ecpg/preproc/type.c#L546-L548
Is this ok?

Reply:
I didn't understand your point.

Reply:
> Do these places require `&& ind_name != NULL` as well?

Yes, sure! I didn't understand it right at first. In those places we also need to add such a condition. Added, thanks!